apps/communications/dialer/js/call_log.js

```diff
@@ -866,12 +867,15 @@ var CallLog = {
     Contacts.findByNumber(phoneNumber,
                           function(contact, matchingTel) {
       if (!contact || !matchingTel) {
-        self._removeContact(log, contactId, updateDb);
+        // Only need one call to remove the contact ID from the call logs.
+        // Use the set to track this.
+        self._removeContact(log, contactId, !self._updateSet.has(contactId));
+        self._updateSet.add(contactId);
```

Comment:
Might be better to make ```_updateSet``` a getter that returns the underlying array, or initializes it if unset.